Python/cp/staircase_problem.py

```diff
@@ -0,0 +1,27 @@
+## This program finds the total number of possible combinations that can be used to
+## climb statirs . EG : for 3 stairs ,combination and output will be 1,1,1 , 1,2 , 2,1 i.e 3 . 
+
+def counting_stairs(stair_number):
+    result = stair_number
+    ##Using Recursion to find the total number of stairs as f(a) = f(a-1) + a(a-2) 
+    ## where f(a) is the final result
+    if(stair_number <=1):
+        result = 1
+    else:
+        result = (counting_stairs(stair_number-1) + counting_stairs(stair_number-2))
+    return result
+
+##Asking for number of steps to find combination.
+count_stair = int(input("Enter total number of stairs: "))
+##passing stair number to  the function.

```

Comment:
The Comments need not to be that detailed. The Code is readable enough. Try to focus on important parts rather than documenting everything.